viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -740,7 +727,7 @@ SLAMServiceImpl::GetLatestPaintedMapSlices() {
             &submap_slice.cairo_data);
     }
     cartographer::io::PaintSubmapSlicesResult painted_slices =
-        viam::io::PaintSubmapSlices(submap_slices, kPixelSize);
+        viam::io::PaintSubmapSlices(submap_slices, resolution);
```

Comment:
[q] if we have an image of 8 pixels with values like

[20, 20, 20, 20, 20, 0, 0, 0, 0]

and want to downsample it to 2 pixels, will it still show some non-zero value for the right-hand pixel?

e.g., either

[20, 5] // use average of underlying pixels

or

[20, 20] // use max of underlying pixels

Reply:
So this is not downsampling the image produced by cairo. Cairo combines the submaps and more accurately their associated probability grids into an image. We then send all the pixels in that image over that arent empty or have a prob of 0. 

So therefore in the image produced by cario has pixel values of [20, 20, 20, 20, 20, 0, 0, 0] we would send over [f(20), f(20), f(20), f(20), f(20)] where the function f() is `calculateViamProbabilityFromColorChannel`.

If instead, you are asking how cairo averages the probability in overlapping submaps/probability grids, the answer is that we are not sure exactly how they are doing this and if they are taking the maximum, average, or some other weighted average

Reply:
As discussed offline, the "cairo combined map -(2)-> cairo combined map using fewer pixels" is a combined step as we send cairo the submap data AND resolution at the same time. The generated painted submaps are the result of that call.